extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -210,97 +293,120 @@ protected function parseImageData($data) {
 
 	private function handlePostAction() {
 		$data = $this->wg->request->getValues();
-		if ( !empty($data) ) {
-			$images = $this->parseImageData($data);
+		if ( !empty( $data ) ) {
+			$images = $this->parseImageData( $data );
 
-			if ( count($images) > 0 ) {
-				$this->helper->updateImageState( $images, $this->action );
+			if ( count( $images ) > 0 ) {
+				$this->imageStateUpdater->update( $images, $this->wg->User->getId() );
 			}
-			$this->ts = null;
-		}
-	}
-
-	private function getOrderingMethod() {
-		if ($this->wg->user->isAllowed( 'imagereviewcontrols' )) {
-			$preferedOrder = (int)$this->app->wg->User->getGlobalPreference( 'imageReviewSort' );
-			$order = $this->request->getInt( 'sort', $preferedOrder );
-
-			if ( $order != $preferedOrder ) {
-				$this->app->wg->User->setGlobalPreference( 'imageReviewSort', $order );
-				$this->app->wg->User->saveSettings();
-			}
-
-			return $order;
+			$this->timestamp = null;
 		}
-		return -1;
 	}
 
 	private function parseAction() {
-		$actions = explode( '/', $this->getPar() );
-		return array_pop( $actions );
+		return array_key_exists( $this->getPar(), self::ALLOWED_ACTIONS ) ? $this->getPar() : self::ACTION_UNREVIEWED;
 	}
 
 	private function checkUserPermissions() {
-		$this->accessQuestionable = $this->wg->User->isAllowed( 'questionableimagereview' );
-		$this->accessRejected = $this->wg->User->isAllowed( 'rejectedimagereview' );
-
 		// check permissions
-		if (!$this->specialPage->userCanExecute($this->wg->User)) {
+		if ( !$this->specialPage->userCanExecute( $this->wg->User ) ) {
 			$this->specialPage->displayRestrictionError();
 			return false;
 		} elseif ( $this->action == self::ACTION_QUESTIONABLE && !$this->accessQuestionable ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 312. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1871)